### PR TITLE
feat(enrolment-type): events enrolment info differs on different enrolment type

### PR DESCRIPTION
**Pull request by nikomakela**
The enrolment information is shown only If the event's enrolment type is internal. If the enrolments are done externally or the event is unenrollable, enrolment status info won't be rendered. In event card also shows the enrolment type.

PT-849 PT-1102.

### CODE REVIEW

File: src/domain/event/EventSummaryPage.tsx [hunk 1]
```diff
@@ -189,7 +193,7 @@ const EventSummaryPage: React.FC = () => {
                       })}
                     </span>
                   </h2>
-                  {!isEventDraft && (
+                  {!isEventDraft && enrolmentType === EnrolmentType.Internal && (
```
Comment (Sartonon) [inline src/domain/event/EventSummaryPage.tsx:196]: minor but result of `enrolmentType === EnrolmentType.Internal` could be saved to a variable just like isEventDraft to make this little bit more concise and easier to read

File: src/domain/event/eventCard/EventCard.tsx [hunk 2]
```diff
@@ -52,6 +55,18 @@ const EventCard: React.FC<Props> = ({
...
+    if (enrolmentType === EnrolmentType.Internal) {
+      return t('events.eventCard.textEnrolments', {
+        count: enrolmentsCount || 0,
+      });
+    } else if (enrolmentType === EnrolmentType.External) {
+      return t('events.eventCard.textExternalEnrolments');
+    } else if (enrolmentType === EnrolmentType.Unenrollable) {
+      return t('events.eventCard.textUnenrollable');
+    }
+  };
```
Comment (Sartonon) [inline src/domain/event/eventCard/EventCard.tsx:68]: Maybe object mapper could be used here to avoid `else if`

Comment (Sartonon) [reply src/domain/event/eventCard/EventCard.tsx:68]: or switch statement

File: src/domain/occurrences/occurrencesTableReadOnly/OccurrencesTableSummary.tsx [hunk 3]
```diff
@@ -44,6 +46,64 @@ const OccurrencesTableSummary: React.FC<Props> = ({
     );
   };
 
+  const enrolmentType = eventData?.event && getEnrolmentType(eventData.event);
+
+  const enrolmentColumns =
+    EnrolmentType.Internal === enrolmentType
+      ? [
```
Comment (Sartonon) [inline src/domain/occurrences/occurrencesTableReadOnly/OccurrencesTableSummary.tsx:53]: To me this seems little bit nasty to use ternary here :D 

how about:

```
let enrolmentColumns = [];
if (EnrolmentType.Internal === enrolmentType) {
  enrolmentColumns = [.....]
}
```

Comment (Sartonon) [reply src/domain/occurrences/occurrencesTableReadOnly/OccurrencesTableSummary.tsx:53]: or if this doesn't break things:

```
  },
    ...(isInternalEnrolment ? enrolmentColumns : []),
    {
```

**Comment by nikomakela** [comment]
Test results: 

It was said that the event does not have enrolments

image

When the occurrences edit form is loaded, it is initialized with internal enrolments (and empty fields)

image

**Pull request merged**